### PR TITLE
TRUNK-3844 Correct BaseOpenmrsObject equals

### PR DESCRIPTION
**Pull request by SandeepRaparthi1**

### CODE REVIEW

**Comment by dkayiwa** [comment]
@SandeepRaparthi1 according to this email thread: https://groups.google.com/a/openmrs.org/forum/#!topic/dev/m0Lfk0Uz3Vc we are going to make changes that will make merging this pull request painful. Do you think you can finish this before Friday next week (27 Feb 2015), to avoid the unnecessary extra work?

File: api/src/main/java/org/openmrs/BaseOpenmrsObject.java [hunk 1]
```diff
@@ -79,6 +82,11 @@ public boolean equals(Object obj) {
 		if (!(obj instanceof BaseOpenmrsObject)) {
 			return false;
 		}
+		//In case of hibernate proxy objects we need to get real classes
+		Class<?> thisClass = Hibernate.getClass(this);
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/BaseOpenmrsObject.java]: Just to check, are we sure that this will work if it is *not* a Hibernate-managed class?

**Comment by SandeepRaparthi1** [comment]
@dkayiwa I have made the changes and commited the code  :) if required any changes please comment will put in my maximum efforts to finish them by next friday thank you :)

**Comment by dkayiwa** [comment]
@SandeepRaparthi1 can you start by answering Darius's question above? :)

File: api/src/main/java/org/openmrs/BaseOpenmrsObject.java (hunk 1 above)
Comment (SandeepRaparthi1) [inline api/src/main/java/org/openmrs/BaseOpenmrsObject.java]: It seems fine and should work :) however i am not not sure :) Please Correct if i am wrong

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/BaseOpenmrsObject.java]: @SandeepRaparthi1 can you create a unit test to confirm it works?

**Comment by SandeepRaparthi1** [comment]
Closing so as to add a new unit test and create a new one :) 
Squash was not possible because local branch was deleted apologies

File: api/src/main/java/org/openmrs/BaseOpenmrsObject.java (hunk 1 above)
Comment (SandeepRaparthi1) [inline api/src/main/java/org/openmrs/BaseOpenmrsObject.java]: @djazayeri as adviced by kayiwa on which non-hibernate managed class you want me to test this on ? :)

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/BaseOpenmrsObject.java]: @SandeepRaparthi1 create an inner class in your unit test.

**Pull request closed without merging**